src/java.base/share/classes/java/lang/Class.java

```diff
@@ -812,14 +812,19 @@ public Void run() {
      *
      * <p> There are nine predefined {@code Class} objects to represent
      * the eight primitive types and void.  These are created by the Java
-     * Virtual Machine, and have the same names as the primitive types that
+     * Virtual Machine, and have the same {@linkplain #getName() names} as the primitive types that
      * they represent, namely {@code boolean}, {@code byte},
      * {@code char}, {@code short}, {@code int},
      * {@code long}, {@code float}, and {@code double}.
```

Comment:
Should we add `{@code void}` to the list here, as this is one of the primitive type names?

Reply:
Hmm. I'll consider that. The javadoc in java.lang.Class is inconsistent in the formatting of "void" as a type name, some instances are in code markup while others are not.

Reply:
@liach please have a look at this intensive exchange between David Holmes (@dholmes-ora) and me: https://github.com/openjdk/jdk/pull/6213#issuecomment-960420076 That exchange relates to other keywords that might appear in plain (i.e. prose) font.

Speaking of which, I might still owe David a follow-up PR.

Reply:
@jddarcy and @pavelrappo, as I understood it, @liach is not suggesting that the term "void" in "primitive types and void" should be replaced by `{@code void}`.

Rather, `{@code void}` should be included in the enumeration of primitive types -- for example, by replacing `and {@code double}` with `{@code double}, and {@code void}`.

Reply:
> @jddarcy and @pavelrappo, as I understood it, @liach is not suggesting that the term "void" in "primitive types and void" should be replaced by `{@code void}`.
> 
> Rather, `{@code void}` should be included in the enumeration of primitive types -- for example, by replacing `and {@code double}` with `{@code double}, and {@code void}`.

AFAIK, that sentence enumerates primitive types, which according to the Java Language Specification [^1][^2], do not include `void`.

[^1]: https://docs.oracle.com/javase/specs/jls/se20/html/jls-4.html#jls-4.2
[^2]: https://docs.oracle.com/javase/specs/jls/se20/html/jls-8.html#jls-8.4.5

Reply:
Thanks for the links, @pavelrappo.

I understand the point you're making; however, there is still ambiguous wording in the documentation for this method.

The method is called `isPrimitive`, and it returns `true` for `isPrimitive(void.class)`. In addition, the first sentence states: "Determines if the specified `Class` object represents a primitive type."

Note as well that the documentation states "There are nine predefined ..." but then goes on to only list 8 names.

So, in line with the intent of the method with regard to `void`, it should be clear to the user that `void` is the name for the type `void.class` (and `Void.TYPE`).

Perhaps the easiest way to include `void.class` and match the 9 predefined types with their names is to remove "primitive" from "as the primitive types that" and replace `and {@code double}` with `{@code double}, and {@code void}`.

Along that line of thinking, it might be best to change the first sentence to "Determines if the specified `Class` object represents a primitive type or void."

Reply:
> Along that line of thinking, it might be best to change the first sentence to "Determines if the specified Class object represents a primitive type or void."

I was just about to suggest exactly that, but from the superinterface method's perspective. Here's what I've been writing when received your message:

At the same, time java.lang.invoke.TypeDescriptor.OfField#isPrimitive:

        /**
         * Does this field descriptor describe a primitive type (including void.)
         *
         * @return whether this field descriptor describes a primitive type
         */
        boolean isPrimitive();

I suggest rephrasing that and similar text elsewhere; for example:

    /** {@return whether this field descriptor describes a primitive type or void} */
    boolean isPrimitive();